packages/fxa-payments-server/src/routes/Product/index.tsx

```diff
@@ -171,6 +172,36 @@ export const Product = ({
 
   const [coupon, setCoupon] = useState<CouponDetails>();
 
+  const [planUpgradeEligibility, setPlanUpgradeEligibility] =
+    useState('invalid');
+
+  const checkPlanUpgradeEligibility = async (planId: string) => {
```

Comment:
nit. To simplify/reduce code, we probably don't need this additional function. We could probably include this logic in the `useEffect` below.